packages/background/src/utils/swaps/1inch.ts

```diff
@@ -106,13 +119,129 @@ export interface OneInchSwapRequestResponse {
+        toToken,
+        amount,
+    }: SwapQuoteParams): OneInchSwapQuoteParams {
+        return {
+            fromTokenAddress:
+                fromToken.address === '0x0'
+                    ? SWAP_NATIVE_ADDRESS
+                    : fromToken.address,
+            toTokenAddress:
+                toToken.address === '0x0'
+                    ? SWAP_NATIVE_ADDRESS
+                    : toToken.address,
+            amount,
+        };
+    },
+    async getSpender(chainId: number): Promise<string> {
+        try {
+            // 1Inch router contract address
+            const res = await retryHandling<OneInchSpenderRes>(() =>
+                httpClient.request<OneInchSpenderRes>(
+                    `${ONEINCH_SWAPS_ENDPOINT}${chainId}/approve/spender`
+                )
+            );
+
+            return res.address;
+        } catch (error) {
+            throw new Error('Unable to fetch exchange spender');
+        }
+    },
+    async getSwapQuote(
+        chainId: number,
+        params: SwapQuoteParams
+    ): Promise<SwapQuoteResponse> {
+        try {
+            const { fromTokenAddress, toTokenAddress, amount } =
+                this.parseQuoteParams(params);
+            const res = await retryHandling<OneInchSwapQuoteResponse>(() =>
+                httpClient.request<OneInchSwapQuoteResponse>(
+                    `${ONEINCH_SWAPS_ENDPOINT}${chainId}/quote`,
+                    {
+                        params: {
+                            fromTokenAddress,
+                            toTokenAddress,
+                            amount,
+                            fee: BASE_SWAP_FEE,
+                        },
+                    }
+                )
+            );
+
+            return {
+                ...res,
+                blockWalletFee: BigNumber.from(res.fromTokenAmount)
+                    .mul(BASE_SWAP_FEE * 10)
+                    .div(1000),
+                estimatedGas: Math.round(res.estimatedGas * GAS_LIMIT_INCREASE),
+            };
+        } catch (error) {
+            const errMessage = map1InchErrorMessage(
+                get1InchErrorMessageFromResponse(error) // Error should be of type RequestError
+            );
+            throw new Error(errMessage || 'Error getting 1Inch swap quote');
+        }
+    },
+    async getSwapParameters(
+        chainId: number,
+        signatureParser: ContractSignatureParser,
+        swapParams: SwapRequestParams
+    ): Promise<SwapParameters> {
+        try {
+            const res = await retryHandling<OneInchSwapRequestResponse>(() =>
+                httpClient.request<OneInchSwapRequestResponse>(
+                    `${ONEINCH_SWAPS_ENDPOINT}${chainId}/swap`,
+                    {
+                        params: {
+                            ...swapParams,
+                            fee: BASE_SWAP_FEE,
+                            referrerAddress: REFERRER_ADDRESS,
+                            allowPartialFill: false,
```

Comment:
the user can't customize this value, right?

Reply:
it's not possible right now. for a matter of better UX we only allowed swapping 100% of the tokens with 1inch. The new default provider (OpenOcean) does not have this option either.